src/PKSim.Core/Services/ProjectSnapshotToSimulationMapper.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using System.Linq;
+using System.Text;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Mappers;
+using OSPSuite.Utility;
+using PKSim.Core.Model;
+using PKSim.Core.Snapshots.Mappers;
+using Project = PKSim.Core.Snapshots.Project;
+
+namespace PKSim.Core.Services;
+
+public interface IProjectSnapshotToSimulationMapper : IMapper<string, IModelCoreSimulation>;
+
+public class ProjectSnapshotToSimulationMapper(
+   IJsonSerializer jsonSerializer,
+   ISnapshotMapper snapshotMapper,
+   ISimulationConfigurationTask simulationConfigurationTask,
+   ISimulationToModelCoreSimulationMapper simulationMapper) : IProjectSnapshotToSimulationMapper
+{
+   public IModelCoreSimulation MapFrom(string snapshotString)
+   {
+      var snapshot = jsonSerializer.DeserializeFromString(Encoding.UTF8.GetString(Convert.FromBase64String(snapshotString)), typeof(Project)).Result as Project;
+      var project = snapshotMapper.MapToModel(snapshot, new ProjectContext(runSimulations: false)).Result as PKSimProject;
+
+      var simulation = project.All<Simulation>().First();
```

Comment:
should we check here that the project contains exactly 1 simulation and throw an explicit exception if not?

Reply:
added